integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/jms/inbound/transport/test/JmsClientAckInboundEndpointTestCase.java

```diff
@@ -0,0 +1,117 @@
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.wso2.carbon.esb.jms.inbound.transport.test;
+
+import org.apache.axiom.om.OMElement;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.automation.engine.annotations.ExecutionEnvironment;
+import org.wso2.carbon.automation.engine.annotations.SetEnvironment;
+import org.wso2.carbon.automation.extensions.servers.jmsserver.client.JMSQueueMessageProducer;
+import org.wso2.carbon.automation.extensions.servers.jmsserver.controller.config.JMSBrokerConfigurationProvider;
+import org.wso2.carbon.integration.common.admin.client.LogViewerClient;
+import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
+import org.wso2.esb.integration.common.utils.Utils;
+import org.wso2.esb.integration.common.utils.servers.ActiveMQServer;
+
+import java.io.File;
+
+/**
+ * This testcase tests consuming message from a queue using client ack mode works
+ */
+public class JmsClientAckInboundEndpointTestCase extends ESBIntegrationTest {
+
+    private static final String QUEUE_NAME = "jmsQueueClientAckInboundEndpointTestCase";
+    private ActiveMQServer activeMQServer = new ActiveMQServer();
+
+    @BeforeClass(alwaysRun = true)
+    public void initialize() throws Exception {
+        activeMQServer.startJMSBroker();
+        super.init();
+
+        verifySequenceExistence("jmsClientAckInboundEPSendInSequence");
+
+        //Add inbound endpoint configuration
+        OMElement inboundEpConfig = esbUtils.loadResource(
+                File.separator + "artifacts" + File.separator + "ESB" + File.separator + "jms" + File.separator +
+                        "inbound" + File.separator + "transport" + File.separator + "jmsClientAckInboundEndpoint.xml");
+        addInboundEndpoint(inboundEpConfig);
+    }
+
+    /**
+     * Test whether consuming message from a queue using client ack mode works
+     *
+     * @throws Exception if any error occurred while running tests
+     */
+    @SetEnvironment(executionEnvironments = {ExecutionEnvironment.STANDALONE})
+    @Test(groups = {"wso2.esb"}, description = "Test JMS client ack mode with inbound endpoint")
+    public void testJmsQueueToHttpWithInboundEndpoint() throws Exception {
+        LogViewerClient logViewerClient = new LogViewerClient(contextUrls.getBackEndUrl(), getSessionCookie());
```

Comment:
Shall we move this to initialize() ?

Reply:
No we can't it needs to be enabled and disabled within the scope of the test method. Other wise there can be unnecessary polling.